### ISSUE TITLE
Shuffling causes fill-extrusion-color/property-function to fail

### ISSUE THREAD

**Issue by brunoabinader** [mapbox/mapbox-gl-native#9978]
Reproduced via:
```
$ node platform/node/test/render.test.js \
    --recycle-map "fill-extrusion-pattern/missing" "fill-extrusion-color/property-function"
* ignore fill-extrusion-pattern missing (https://github.com/mapbox/mapbox-gl-js/issues/3327)
* failed fill-extrusion-color property-function
1 passed but were ignored (50.0%)
1 failed (50.0%)
```

| expected | actual | diff |
|--|--|--|
| ![expected](https://user-images.githubusercontent.com/76133/30339026-503837f0-97f6-11e7-81f4-d6c0f0551083.png) | ![actual](https://user-images.githubusercontent.com/76133/30339017-4a20990c-97f6-11e7-98d9-48231f336f9b.png) | ![diff](https://user-images.githubusercontent.com/76133/30339043-5c56cd76-97f6-11e7-8077-e4280ff5fc78.png) |

It seems the previous _faulty_ `fill-extrusion-pattern/missing` run is messing with the layer z-ordering in `fill-extrusion-color/property-function`.

/cc @lbud

**Comment by lbud** [comment]
I started digging into this and realized something extra weird was going on where only sometimes, and only (afaict) when using fill-extrusion-pattern layers, the extrusion texture was being bound without a depth buffer. This can happen with missing patterns (like in this case) or with valid patterns, with other layers or without — my best guess is that something in the ImageManager binding is interfering with the state. I haven't figured out exactly what it is yet, BUT since https://github.com/mapbox/mapbox-gl-native/pull/9931 majorly overhauls the way create/use offscreen textures for extrusions, I tested several cases/platforms where I was reproducing this bug on that branch and it seems to fix them. 🤷‍♀️
